lib/rules/selector-pseudo-class-no-unknown/index.js

```diff
@@ -68,8 +69,13 @@ const rule = (primary, secondaryOptions) => {
 						return;
 					}
 
+					const name = value.replace(/^:*/, '').toLowerCase();
+					const isPseudoElement =
+						value.slice(0, 2) === '::' || levelOneAndTwoPseudoElements.has(name);
+					const hasVendorPrefix = vendor.prefix(name);
+
 					// Ignore pseudo-elements
-					if (value.slice(0, 2) === '::') {
+					if (isPseudoElement) {
 						return;
 					}
```

Comment:
[suggest] For performance, can you evaluate expressions lazily as possible like this?

```js
if (value.startsWith('::')) {
	return;
}

const name = value.replace(/^:*/, '').toLowerCase();

if (levelOneAndTwoPseudoElements.has(name)) {
	return;
}

if (optionsMatches(secondaryOptions, 'ignorePseudoClasses', pseudoNode.value.slice(1))) {
	return;
}

const hasVendorPrefix = vendor.prefix(name);
let index = null;
```